### PR TITLE
Add matrix overflow test cases.

### PR DESCRIPTION
**Pull request by dj2**
This CL tests that const-eval of matric operations cause over/under flow.

Fixes: #3682 

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [x] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/validation/expression/matrix/add_sub.spec.ts [hunk 1]
```diff
@@ -163,3 +164,113 @@ fn main() {
...
+fn main() {
+  const foo = ${lhs} + ${rhs};
+}
+`;
+
+    t.expectCompileResult(t.params.rhs === 1, code);
+  });
+
+g.test('underflow,f16')
+  .desc(`Validates that f16 add overflows in shader creation`)
```
Comment (zoddicus) [inline src/webgpu/shader/validation/expression/matrix/add_sub.spec.ts:224]: nit: sub underflows

File: src/webgpu/shader/validation/expression/matrix/add_sub.spec.ts [hunk 2]
```diff
@@ -163,3 +164,113 @@ fn main() {
...
+fn main() {
+  const foo = ${lhs} + ${rhs};
+}
+`;
+
+    t.expectCompileResult(t.params.rhs === 1, code);
+  });
+
+g.test('underflow,abstract')
+  .desc(`Validates that abstract add overflows in shader creation`)
```
Comment (zoddicus) [inline src/webgpu/shader/validation/expression/matrix/add_sub.spec.ts:262]: nit: sub underflows

**Review by zoddicus — COMMENTED** [review]
The general approach seems sound, a few comment nits.

You are only testing 2x2 matrices. @dneto0 thoughts on if various matrix shapes need to be iterated

File: src/webgpu/shader/validation/expression/matrix/add_sub.spec.ts [hunk 3]
```diff
@@ -163,3 +164,113 @@ fn main() {
...
+fn main() {
+  const foo = ${lhs} + ${rhs};
+}
+`;
+
+    t.expectCompileResult(t.params.rhs === 1, code);
+  });
+
+g.test('underflow,f32')
+  .desc(`Validates that f32 add overflows in shader creation`)
```
[the remark this answers is not in the document]
Comment (dj2) [reply src/webgpu/shader/validation/expression/matrix/add_sub.spec.ts]: done

File: src/webgpu/shader/validation/expression/matrix/add_sub.spec.ts (hunk 2 above)
Comment (dj2) [reply src/webgpu/shader/validation/expression/matrix/add_sub.spec.ts]: done

File: src/webgpu/shader/validation/expression/matrix/add_sub.spec.ts (hunk 1 above)
Comment (dj2) [reply src/webgpu/shader/validation/expression/matrix/add_sub.spec.ts]: done

**Comment by dj2** [comment]
Yea, I wasn't sure if it was worth doing more, seemed like if you're const eval-ing one, you'd be doing them all.

**Comment by dj2** [comment]
I extended the tests to run over all the matrix sizes. I've also added a set of tests where the overflow happens during the internal addition instead of the initial multiplication.

**Review by zoddicus — APPROVED** [review]
